### ISSUE TITLE
Application-Master Interface: Sequence Diagram

### ISSUE THREAD

**Issue by alyhasansakr** [EdgeLab-FHDO/Edge-Diagnostic-Platform#21]
Develop a simple sequence diagram for the communication between an application and the master.
You may refer to the old code, search the Internet or just make your own.
This is a discussion topic, there is no right answer here.

**Comment by juan-castrillon** [comment]
Before I start the modeling, I would like to see if the base requirements I will base it on are correct:

- The client (application), should be able to ask for a node (server) to the master and get one
-  The client should send the master the limits for the node(s) it has (for the master to use resource limiting)
- The master should be able to control the application (via events) [start, stop, pause, move]

- This one is not quite clear, Should the application send logging data to the master? or to a central logger, like we discussed in #28 ? 

Am I missing anything? What do you think? @theodorelu27 , @alyhasansakr

**Comment by alyhasansakr** [comment]
@juan-castrillon 
It is always good to review the requirements before starting.
- About the second point, I believe resource limiting should be done through events, the client has nothing to do with it. We assume that some sort of evaluation was done to the client and the server to estimate the used and available resources, this information is used directly by the master as part of a scenario.
- About the last point, the application should send the logging data to a central logger, I would say it is not part of this issue.
- I think you are missing the part after the client receives a node, the client should inform the master about its status with the node (basically, connected and disconnected).
